### PR TITLE
[PPP-4660] - Use of Vulnerable Component: Components/jackson-databind:2.9.10.2

### PR DESCRIPTION
**Pull request by andreramos89**
@smmribeiro 
@bcostahitachivantara 
@renato-s

### CODE REVIEW

**Comment by buildguy** [comment]
### :x: Build failed in 1h 15m 3s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl extensions,user-console
```
---
:ok_hand: All tests passed!
 
Tests run: **1664**, Failures: **0**, Skipped: **1** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/platform-11/job/PR-5182/1/testReport)

 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
script returned exit code 1
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 45m 40s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl extensions,user-console
```
---
:ok_hand: All tests passed!
 
Tests run: **1664**, Failures: **0**, Skipped: **1** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/platform-11/job/PR-5182/2/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 44m 48s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl extensions,user-console
```
---
:ok_hand: All tests passed!
 
Tests run: **1664**, Failures: **0**, Skipped: **1** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/platform-11/job/PR-5182/3/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :x: Build failed in 39s
Build command:
```console
mvn clean verify -B -e -Daudit -amd
```
---
 :exclamation: No tests found!
 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
script returned exit code 1
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_

File: extensions/src/main/java/org/pentaho/platform/web/http/api/resources/MantleResource.java [hunk 1]
```diff
@@ -54,10 +59,10 @@ public class MantleResource {
   @GET
   @Path( "/getDeeplinkAllowedHosts" )
   @Produces( { APPLICATION_JSON } )
-  @StatusCodes( { @ResponseCode( code = 200, condition = "Successfully sent result." ),
+  @StatusCodes( { @ResponseCode( code = 200, condition = "Successfully obtained allowed host list." ),
     @ResponseCode( code = 500, condition = "Failed to get data." ) } )
-  public List<String> getDeeplinkAllowedHosts() {
-    return getListHosts();
+  public Response getDeeplinkAllowedHosts() {
+    return Response.ok(  new HostsWrapper( getListHosts() ), MediaType.APPLICATION_JSON ).build();
```
[the remark this answers is not in the document]
Comment (andreramos89) [reply extensions/src/main/java/org/pentaho/platform/web/http/api/resources/MantleResource.java]: A new push has been done

**Review by smmribeiro — APPROVED** [review]
Looks good.

File: extensions/src/main/java/org/pentaho/platform/web/http/api/resources/HostsWrapper.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+  List<String> hosts = new ArrayList<String>();
+
+  public HostsWrapper() {
+  }
+
+  public HostsWrapper(List<String> hosts ) {
+    this.hosts.addAll( hosts );
+  }
+
```
Comment (NJtwentyone) [inline extensions/src/main/java/org/pentaho/platform/web/http/api/resources/HostsWrapper.java:39]: this class needs unit tests

File: assemblies/pentaho-war/pom.xml [hunk 3]
```diff
@@ -11,7 +11,6 @@
   <version>9.5.0.0-SNAPSHOT</version>
   <packaging>war</packaging>
   <properties>
-    <snakeyaml.version>1.13</snakeyaml.version>
```
Comment (NJtwentyone) [inline assemblies/pentaho-war/pom.xml:14]: you can't delete this property without replace/updating it in other places. You are getting a build error now

https://wingman.orl.eng.hitachivantara.com/blue/organizations/jenkins/projects%2Fplatform-11/detail/PR-5182/5/pipeline


`${snakeyaml.version} `is used in:

- ./repository/pom.xml:190:      <version>${snakeyaml.version}</version>
- ./assemblies/pentaho-war/pom.xml:860:      <version>${snakeyaml.version}</version>
- ./extensions/pom.xml:1851:      <version>${snakeyaml.version}</version>

File: user-console/src/main/java/org/pentaho/mantle/client/MantleApplication.java [hunk 4]
```diff
@@ -20,10 +20,7 @@
 
 package org.pentaho.mantle.client;
 
-import com.google.gwt.core.client.GWT;
-import com.google.gwt.core.client.JavaScriptObject;
-import com.google.gwt.core.client.JsArray;
-import com.google.gwt.core.client.JsonUtils;
+import com.google.gwt.core.client.*;
```
Comment (NJtwentyone) [inline user-console/src/main/java/org/pentaho/mantle/client/MantleApplication.java:23]: Dont't use star imports `.*.` your penatho checktyle should have prevented this.

File: extensions/src/main/java/org/pentaho/platform/web/http/api/resources/MantleResource.java [hunk 5]
```diff
@@ -49,32 +52,34 @@
...
+  HostsWrapper getWrappedResponse() {
+    return new HostsWrapper(getListHosts());
   }
 
   private List<String> getListHosts() {
     if ( whiteListedHosts.isEmpty() ) {
 
-      File propFile = new File(
-        PentahoSystem.getApplicationContext().getSolutionRootPath() + "/system/deeplink_allowed_urls.properties" );
+      getPropFile();
```
Comment (NJtwentyone) [inline extensions/src/main/java/org/pentaho/platform/web/http/api/resources/MantleResource.java:78]: I'm confused about this line.
`getPropFile();`

Before the values of file deeplink_allowed_urls.properties were used inside this function. But with the proposed changes, this file isn't used.

And why call a function that the results are never used?

File: user-console/src/main/java/org/pentaho/mantle/client/MantleApplication.java [hunk 6]
```diff
@@ -495,11 +492,14 @@ private void loadWhitelistedHosts() {
       requestBuilder.sendRequest( null, new RequestCallback() {
         @Override public void onResponseReceived( Request request, Response response ) {
           List<String> whitelistedHosts = new ArrayList<>();
-          JSONArray data = new JSONArray( JsonUtils.safeEval( response.getText() ) );
-          for ( int i = 0; i < data.size(); i++ ) {
+
```
Comment (NJtwentyone) [inline user-console/src/main/java/org/pentaho/mantle/client/MantleApplication.java:495]: You'll need test for MantleApplication as well. You can use this as an reference https://github.com/pentaho/pentaho-platform/blob/master/extensions/src/test/java/org/pentaho/platform/web/http/api/resources/EmailResourceTest.java

You won't have to do the MicroPlatform logic in #setup 

- https://github.com/pentaho/pentaho-platform/blob/master/extensions/src/test/java/org/pentaho/platform/web/http/api/resources/EmailResourceTest.java#L50-L52

Also, you won;t have to do the KettleEnvironment logic either

- https://github.com/pentaho/pentaho-platform/blob/master/extensions/src/test/java/org/pentaho/platform/web/http/api/resources/EmailResourceTest.java#L59
- https://github.com/pentaho/pentaho-platform/blob/master/extensions/src/test/java/org/pentaho/platform/web/http/api/resources/EmailResourceTest.java#L92

**Review by NJtwentyone — CHANGES_REQUESTED** [review]
Main points

- build error
- pentaho coding standards issues

**Comment by buildguy** [comment]
### :x: Build failed in 1m 21s
Build command:
```console
mvn clean verify -B -e -Daudit -amd
```
---
 :exclamation: No tests found!
 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
script returned exit code 1
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_

**Comment by smmribeiro** [comment]
Closing this and all "January 2023" related PRs for this issue (check this [comment](https://hv-eng.atlassian.net/browse/PPP-4660?focusedCommentId=1546417) and this [comment](https://hv-eng.atlassian.net/browse/PPP-4660?focusedCommentId=1543281)).

Too much time has passed and the risk of a PR being incomplete (or simply wrong) is too high.
I won't delete the original branch so that anyone picking up the development can see what was done previously.

**Comment by smmribeiro** [comment]
Closing this and all "January 2023" related PRs for this issue (check this [comment](https://hv-eng.atlassian.net/browse/PPP-4660?focusedCommentId=1546417) and this [comment](https://hv-eng.atlassian.net/browse/PPP-4660?focusedCommentId=1543281)).

Too much time has passed and the risk of a PR being incomplete (or simply wrong) is too high.
I won't delete the original branch so that anyone picking up the development can see what was done previously.

**Pull request closed without merging**